File: frontend/src/services/guidance.ts
import { api } from "./api";
import type { GuidanceRequest, GuidanceResponse } from "@/types/guidance";

export async function getGuidance(
  repositoryId: string,
  task: string
): Promise<GuidanceResponse> {
  const payload: GuidanceRequest = { task };
  return api.post<GuidanceResponse>(
    `/repositories/${repositoryId}/guidance`,
    payload
  );
}

File: backend/app/schemas/guidance.py
"""Developer guidance API schemas."""

from pydantic import BaseModel, Field


class GuidanceRequest(BaseModel):
    task: str = Field(..., min_length=3)


class GuidanceStep(BaseModel):
    title: str
    files: list[str]
    reasoning: str


class GuidanceResponse(BaseModel):
    task: str
    steps: list[GuidanceStep]
